core/libp2p/transport/tcp/tcp_connection.cpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/transport/tcp/tcp_connection.hpp"
+
+#include <sstream>
+
+namespace libp2p::transport {
+
+  TcpConnection::TcpConnection(boost::asio::io_context &context)
+      : socket_(context) {}
+
+  TcpConnection::TcpConnection(boost::asio::ip::tcp::socket socket)
+      : socket_(std::move(socket)) {}
+
+  outcome::result<multi::Multiaddress> TcpConnection::getRemoteMultiaddr()
+      const {
+    try {
+      auto &&remote = socket_.remote_endpoint();
+      auto &&addr = remote.address().to_v4().to_string();
+      auto &&port = remote.port();
+      std::ostringstream s{};
+      s << "/ip4/" << addr << "/tcp/" << port;
+      OUTCOME_TRY(ma, multi::Multiaddress::create(s.str()));
+      return ma;
```

Comment:
Maybe save the result to the local variable, as it can be accessed several times, and this overhead each time is not good?

Reply:
it is a memory/cpu tradeoff. I don't think you need to access this method many times in a loop